config/dev/services.yml

```diff
@@ -0,0 +1,12 @@
+imports:
+  - { resource: '../prod/services.yml' }
+
+services:
+  Symfony\Component\Console\Output\BufferedOutput: ~
```

Comment:
This is only used during testing. It’s necessary because during testing, we use BufferedOutput instead of the ConsoleOutput. The same issue occurs for input--we need to switch from ArgvInput to ArrayInput.

Without this services file override, services will be auto wired incorrectly during testing.  We would end up with multiple inputs and outputs of different types. For example, the production services file would create an ArgvInput and a ConsoleOutput for all services accepting InputInterface and OutputInterface arguments, like the ExceptionListener and the TelemetryHelper.

Our test would then create a separate ArrayInput and BufferedOutput and inject those into the Application. At that point the application would be using a different input and output than the other services in the container. This causes all sorts of problems, not least of which is our inability to capture and assert the value of the output of services.

With the currently proposed approach, we correctly inject these services into the container and obviate the need to create them manually. We also ensure that there is only one input and one output object and that those objects are shared across services.

Reply:
I've actually been trying to figure out how to do this for a while. I made an attempt in https://github.com/acquia/cli/pull/326, but it was really just a workaround. We still need to refactor TestBase: https://github.com/acquia/cli/blob/8d03cc72d75aa127a8c7fa82cb2936edb506403b/tests/phpunit/src/TestBase.php#L224-L231

The container should autowire itself during testing, this type of manually input and outputting per service isn't quite right.